# ITMGT 25.03 EE
# Assignment 3A

This is a data-cleaning assignment and is the first of two parts. The second part (3B) will be given Monday or Tuesday next week. This notebook is uploaded to our Github repository along with a csv of fake, dirty data called **recweekdata.csv**.


The output of this assignment is to convert recweekdata.csv into processable data for statistics. Submit the following files through GitHub:
1. A cleaned recweekdata.csv called cleanrecweekdata.csv **(5 points)**
2. This notebook, containing the code (properly commented) that you used to clean the data **(5 points)**
3. An html file of summary statistics from each column called **stats.html**. The requirements for the summary statistics are described below. **(10 points)**

This entire assignment (3A) is worth 20 points.

Upload your output files to your Github repositories. Make sure to send an email to **joben.ilagan@gmail.com** once you're done.

This assignment is due on **Tuesday, July 9, 2019** at 12:00NN.

In [2]:
# To study the contents of the dirty information, finding commonalities as a means to cleaning. 

import csv

with open("recweekdata.csv") as dirty_file:
    dirty_data = csv.reader(dirty_file, delimiter = ",")
    line = 0
    for row in dirty_data:
        if line == 0:
            pass
        else:
            print(row)
        line+=1

['name1', 'lastname1', '11', '26', '1998', 'BS LM', '+63 918 302 6210']
['name2', 'lastname2', 'May', '26', '2001', 'Management', '927 212 6276']
['name3', 'lastname3', '5', '16', '1997', 'BS CTM', '+63 918 253 7017']
['name4', 'lastname4', 'Dec', '9', '97', 'BS ITE', '918 714 4360']
['name5', 'lastname5', 'Sep', '5', '97', 'BS CTM', '+63 927 462 1674']
['name6', 'lastname6', 'March', '2', '1999', 'Management', '+63 918 629 5268']
['name7', 'lastname7', '11', '10', '98', 'BS LM', '+63 917 369 7388']
['name8', 'lastname8', 'Aug', '28', '99', 'Management Engineering', '917 240 6000']
['name9', 'lastname9', 'July', '10', '97', 'Management Engineering', '918 226 5748']
['name10', 'lastname10', '9', '14', '2001', 'CTM', '+63 918 342 1717']
['name11', 'lastname11', 'December', '8', '98', 'BS LM', '+63 927 708 1308']
['name12', 'lastname12', '5', '4', '97', 'BS ME', '927 747 5497']
['name13', 'lastname13', 'January', '28', '1999', 'BS CTM', '+63 917 959 4490']
['name14', 'lastname14', '1', '1

['name1663', 'lastname1663', 'June', '7', '99', 'Information Technology Entrepreneurship', '+63 927 282 8654']
['name1664', 'lastname1664', 'June', '9', '2001', 'BS LM', '+63 917 109 2041']
['name1665', 'lastname1665', 'March', '27', '1998', 'Management', '+63 918 876 9262']
['name1666', 'lastname1666', 'Oct', '4', '2000', 'Legal Management', '+63 917 493 5680']
['name1667', 'lastname1667', 'September', '25', '99', 'LM', '917 782 2046']
['name1668', 'lastname1668', '8', '2', '1998', 'CTM', '918 316 8589']
['name1669', 'lastname1669', '6', '16', '1997', 'BS LM', '927 927 6310']
['name1670', 'lastname1670', '7', '19', '1997', 'Management Engineering', '927 401 8816']
['name1671', 'lastname1671', 'February', '27', '2001', 'BS ITE', '917 234 2366']
['name1672', 'lastname1672', '5', '9', '1999', 'BS LM', '917 211 9349']
['name1673', 'lastname1673', 'Jan', '8', '2000', 'Communications Technology', '917 528 4529']
['name1674', 'lastname1674', 'Nov', '19', '98', 'ME', '+63 917 209 1835']
['nam

['name2980', 'lastname2980', 'July', '3', '99', 'BS LM', '918 634 6229']
['name2981', 'lastname2981', 'Feb', '17', '1998', 'Information Technology Entrepreneurship', '+63 927 393 1660']
['name2982', 'lastname2982', '3', '28', '2000', 'Information Technology Entrepreneurship', '+63 927 642 3573']
['name2983', 'lastname2983', 'December', '3', '98', 'BS CTM', '918 106 5583']
['name2984', 'lastname2984', 'Nov', '13', '2000', 'BS MGT', '+63 917 612 2550']
['name2985', 'lastname2985', 'Feb', '23', '99', 'BS MGT', '927 108 3318']
['name2986', 'lastname2986', '3', '2', '1999', 'ITE', '927 622 4725']
['name2987', 'lastname2987', 'Oct', '6', '97', 'ITE', '+63 918 289 5164']
['name2988', 'lastname2988', 'Feb', '20', '1997', 'BS MGT', '+63 927 984 2369']
['name2989', 'lastname2989', 'Jan', '17', '1999', 'BS ME', '927 909 9560']
['name2990', 'lastname2990', 'Jul', '14', '97', 'Management', '927 350 9410']
['name2991', 'lastname2991', 'July', '27', '2000', 'Information Technology Entrepreneurship', 

In [3]:
# Formulation of individual rows into dictionary equivalents

import csv

with open("recweekdata.csv") as dirty_file:
    dirty_data = csv.reader(dirty_file, delimiter = ",")
    recweek_dict = dict()
    line = 0
    for row in dirty_data:
        if line == 0:
            pass
        else:
            recweek_dict[(row[0])[4:]] = {"First Name": row[0], "Last Name": row[1], "Month": row[2], "Day": row[3], "Year": row[4], "Course": row[5], "Phone Number": row[6]}
        line += 1
    print(recweek_dict)
              

{'1': {'First Name': 'name1', 'Last Name': 'lastname1', 'Month': '11', 'Day': '26', 'Year': '1998', 'Course': 'BS LM', 'Phone Number': '+63 918 302 6210'}, '2': {'First Name': 'name2', 'Last Name': 'lastname2', 'Month': 'May', 'Day': '26', 'Year': '2001', 'Course': 'Management', 'Phone Number': '927 212 6276'}, '3': {'First Name': 'name3', 'Last Name': 'lastname3', 'Month': '5', 'Day': '16', 'Year': '1997', 'Course': 'BS CTM', 'Phone Number': '+63 918 253 7017'}, '4': {'First Name': 'name4', 'Last Name': 'lastname4', 'Month': 'Dec', 'Day': '9', 'Year': '97', 'Course': 'BS ITE', 'Phone Number': '918 714 4360'}, '5': {'First Name': 'name5', 'Last Name': 'lastname5', 'Month': 'Sep', 'Day': '5', 'Year': '97', 'Course': 'BS CTM', 'Phone Number': '+63 927 462 1674'}, '6': {'First Name': 'name6', 'Last Name': 'lastname6', 'Month': 'March', 'Day': '2', 'Year': '1999', 'Course': 'Management', 'Phone Number': '+63 918 629 5268'}, '7': {'First Name': 'name7', 'Last Name': 'lastname7', 'Month': '1

In [4]:
# Cleaning the Birth Month

import re

for row in range (1, len(recweek_dict)+1,1):
    # print(recweek_dict[str(row)]["Month"])
    if len(str(recweek_dict[str(row)]["Month"])) == 1:
        recweek_dict[str(row)]["Month"] = re.sub(r'(9)',r'September',recweek_dict[str(row)]["Month"])
        recweek_dict[str(row)]["Month"] = re.sub(r'(8)',r'August',recweek_dict[str(row)]["Month"])
        recweek_dict[str(row)]["Month"] = re.sub(r'(7)',r'July',recweek_dict[str(row)]["Month"])
        recweek_dict[str(row)]["Month"] = re.sub(r'(6)',r'June',recweek_dict[str(row)]["Month"])
        recweek_dict[str(row)]["Month"] = re.sub(r'(5)',r'May',recweek_dict[str(row)]["Month"])
        recweek_dict[str(row)]["Month"] = re.sub(r'(4)',r'April',recweek_dict[str(row)]["Month"])
        recweek_dict[str(row)]["Month"] = re.sub(r'(3)',r'March',recweek_dict[str(row)]["Month"])
        recweek_dict[str(row)]["Month"] = re.sub(r'(2)',r'February',recweek_dict[str(row)]["Month"])
        recweek_dict[str(row)]["Month"] = re.sub(r'(1)',r'January',recweek_dict[str(row)]["Month"]) # Tried limiting the range
    elif len(str(recweek_dict[str(row)]["Month"])) == 2:   
        recweek_dict[str(row)]["Month"] = re.sub(r'(12)',r'December',recweek_dict[str(row)]["Month"]) # Tried positional by putting this first so as to read 12 first
        recweek_dict[str(row)]["Month"] = re.sub(r'(11)',r'November',recweek_dict[str(row)]["Month"])
        recweek_dict[str(row)]["Month"] = re.sub(r'(10)',r'October',recweek_dict[str(row)]["Month"])
    elif len(str(recweek_dict[str(row)]["Month"])) == 3:
        recweek_dict[str(row)]["Month"] = re.sub(r'(Dec)',r'December',recweek_dict[str(row)]["Month"])
        recweek_dict[str(row)]["Month"] = re.sub(r'(Nov)',r'November',recweek_dict[str(row)]["Month"])
        recweek_dict[str(row)]["Month"] = re.sub(r'(Oct)',r'October',recweek_dict[str(row)]["Month"])
        recweek_dict[str(row)]["Month"] = re.sub(r'(Sep)',r'September',recweek_dict[str(row)]["Month"])
        recweek_dict[str(row)]["Month"] = re.sub(r'(Aug)',r'August',recweek_dict[str(row)]["Month"])
        recweek_dict[str(row)]["Month"] = re.sub(r'(Jul)',r'July',recweek_dict[str(row)]["Month"])
        recweek_dict[str(row)]["Month"] = re.sub(r'(Jun)',r'June',recweek_dict[str(row)]["Month"])
        recweek_dict[str(row)]["Month"] = re.sub(r'(5)',r'May',recweek_dict[str(row)]["Month"])
        recweek_dict[str(row)]["Month"] = re.sub(r'(Apr)',r'April',recweek_dict[str(row)]["Month"])
        recweek_dict[str(row)]["Month"] = re.sub(r'(Mar)',r'March',recweek_dict[str(row)]["Month"])
        recweek_dict[str(row)]["Month"] = re.sub(r'(Feb)',r'February',recweek_dict[str(row)]["Month"])
        recweek_dict[str(row)]["Month"] = re.sub(r'(Jan)',r'January',recweek_dict[str(row)]["Month"])
    else: 
        pass
    print(recweek_dict[str(row)]["Month"])

# Works perfectly well, but rigid hard coding.


November
May
May
December
September
March
November
August
July
September
December
May
January
January
June
April
January
April
December
March
July
September
April
October
May
December
March
November
October
May
April
July
December
October
July
December
February
August
May
February
November
September
January
February
January
December
August
April
October
March
August
March
December
July
May
September
May
April
April
August
September
December
February
November
March
September
December
November
November
May
August
December
June
October
September
February
May
September
December
August
February
November
June
February
January
November
August
January
March
March
August
December
June
November
September
March
January
January
May
May
April
August
December
July
March
August
April
May
August
January
December
May
May
August
March
July
January
February
December
February
March
March
April
February
November
June
March
June
October
July
February
September
April
May
October
October
June
February
April
S

April
August
August
December
December
October
April
April
July
June
May
February
February
January
December
November
July
April
January
December
June
December
January
November
July
November
November
February
February
November
June
April
May
June
February
December
September
November
June
April
December
August
November
August
September
April
May
October
December
December
June
September
March
September
March
October
February
June
December
November
April
August
June
January
May
October
August
August
April
January
November
July
March
August
February
June
April
April
June
April
February
October
July
March
April
September
May
February
November
January
August
September
May
March
November
April
May
July
December
March
July
March
April
November
December
November
February
April
October
November
December
June
June
October
September
October
January
June
April
September
December
May
November
September
October
June
February
January
October
March
November
February
January
April
September
August
July
Ju

January
May
June
March
September
August
July
July
May
December
September
November
September
December
June
December
March
July
March
August
March
July
August
November
September
May
January
October
March
January
May
April
June
January
June
November
February
April
August
May
April
July
March
November
July
January
April
January
January
January
November
September
December
January
August
June
May
January
March
May
October
July
December
July
June
August
August
August
October
February
December
November
November
December
August
August
July
November
May
May
June
March
May
January
September
October
February
April
January
May
January
February
December
January
September
May
November
December
June
October
July
July
September
June
January
March
September
July
March
September
January
July
September
February
July
March
April
July
April
December
February
December
January
September
November
November
November
March
August
February
February
March
May
December
May
December
February
August
January
February
J

In [5]:
# Cleaning the Birth Year

import re

for row in range (1, len(recweek_dict)+1,1):
    #print(recweek_dict[str(row)]["Year"])
    if len(str(recweek_dict[str(row)]["Year"])) == 2:
        recweek_dict[str(row)]["Year"] = re.sub(r'01',r'2001',recweek_dict[str(row)]["Year"])
        recweek_dict[str(row)]["Year"] = re.sub(r'00',r'2000',recweek_dict[str(row)]["Year"])
        recweek_dict[str(row)]["Year"] = re.sub(r'99',r'1999',recweek_dict[str(row)]["Year"])
        recweek_dict[str(row)]["Year"] = re.sub(r'98',r'1998',recweek_dict[str(row)]["Year"])
        recweek_dict[str(row)]["Year"] = re.sub(r'97',r'1997',recweek_dict[str(row)]["Year"])
    else: 
        pass
    print(recweek_dict[str(row)]["Year"])
    
# Works Completely well.


1998
2001
1997
1997
1997
1999
1998
1999
1997
2001
1998
1997
1999
1999
1998
1999
1997
1997
1997
1997
1999
1999
1997
1997
1997
1999
1997
2000
1999
1999
1999
1999
1999
1999
2001
1998
2000
2001
1997
1998
1999
1998
1999
1998
1997
1998
1999
1999
1997
2001
1998
1999
2000
1998
1998
1998
2001
1997
1998
1999
1998
2000
2000
1997
1997
1997
1997
2001
1998
1998
1999
1999
1998
1997
1998
1997
2000
2000
2000
1999
1997
1997
2000
1999
1999
1998
1998
1997
1998
1999
1997
1997
1998
1997
2001
1999
1997
1998
1998
1999
1997
2001
2000
2001
1999
1997
2001
1999
1998
1999
2001
1997
1997
1999
1998
1997
1997
1999
1997
1997
2000
2001
1999
1999
1998
2000
2000
1998
1998
2000
1999
1999
1999
2000
1999
2000
1998
1997
1999
1999
1998
1999
1999
1998
2001
1998
1999
2000
1997
1997
1998
1999
1999
1999
1999
1998
1997
1999
1998
2000
2000
2000
1999
1997
1998
1999
2000
1997
1998
2000
1997
2000
1998
1998
1999
1999
2001
2001
1998
1999
1999
1998
1997
1998
1997
1997
2001
1997
2000
1997
1999
1999
1997
1998
2001
2001
2001
1997
1999
1998


1998
1999
1997
1999
1998
2000
1998
2000
1999
1999
1997
1997
1999
1997
2000
1999
2000
1999
1997
1997
1998
1998
1998
1997


In [6]:
# Cleaning the Course

import re

for row in range (1, len(recweek_dict)+1,1):
    #print(recweek_dict[str(row)]["Course"])
    if recweek_dict[str(row)]["Course"] == "BS Management":
        continue
    elif recweek_dict[str(row)]["Course"] == "BS Management Engineering":
        continue
    elif recweek_dict[str(row)]["Course"] == "BS IT Entrepreneurship":
        continue
    elif recweek_dict[str(row)]["Course"] == "BS Legal Management":
        continue
    elif recweek_dict[str(row)]["Course"] == "BS Communications Technology":
        continue
    elif recweek_dict[str(row)]["Course"] == "Management":
        recweek_dict[str(row)]["Course"] = re.sub(r'Management',r'BS Management',recweek_dict[str(row)]["Course"])
    elif recweek_dict[str(row)]["Course"] == "Management Engineering":
        recweek_dict[str(row)]["Course"] = re.sub(r'Management Engineering',r'BS Management Engineering',recweek_dict[str(row)]["Course"])
    elif recweek_dict[str(row)]["Course"] == "Information Technology Entrepreneurship":
        recweek_dict[str(row)]["Course"] = re.sub(r'Information Technology Entrepreneurship',r'BS IT Entrepreneurship',recweek_dict[str(row)]["Course"])
    elif recweek_dict[str(row)]["Course"] == "Legal Management":
        recweek_dict[str(row)]["Course"] = re.sub(r'Legal Management',r'BS Legal Management',recweek_dict[str(row)]["Course"])
    elif recweek_dict[str(row)]["Course"] == "Communications Technology":
        recweek_dict[str(row)]["Course"] = re.sub(r'Communications Technology',r'BS Communications Technology',recweek_dict[str(row)]["Course"])
    elif len(recweek_dict[str(row)]["Course"]) <= 3:
        recweek_dict[str(row)]["Course"] = re.sub(r'MGT',r'BS Management',recweek_dict[str(row)]["Course"])
        recweek_dict[str(row)]["Course"] = re.sub(r'ME',r'BS Management Engineering',recweek_dict[str(row)]["Course"])
        recweek_dict[str(row)]["Course"] = re.sub(r'ITE',r'BS IT Entrepreneurship',recweek_dict[str(row)]["Course"])
        recweek_dict[str(row)]["Course"] = re.sub(r'LM',r'BS Legal Management',recweek_dict[str(row)]["Course"])
        recweek_dict[str(row)]["Course"] = re.sub(r'CTM',r'BS Communications Technology',recweek_dict[str(row)]["Course"])
    elif 5 <= len(recweek_dict[str(row)]["Course"]) <= 6:
        recweek_dict[str(row)]["Course"] = re.sub(r'BS MGT',r'BS Management',recweek_dict[str(row)]["Course"])
        recweek_dict[str(row)]["Course"] = re.sub(r'BS ME',r'BS Management Engineering',recweek_dict[str(row)]["Course"])
        recweek_dict[str(row)]["Course"] = re.sub(r'BS ITE',r'BS IT Entrepreneurship',recweek_dict[str(row)]["Course"])
        recweek_dict[str(row)]["Course"] = re.sub(r'BS LM',r'BS Legal Management',recweek_dict[str(row)]["Course"])
        recweek_dict[str(row)]["Course"] = re.sub(r'BS CTM',r'BS Communications Technology',recweek_dict[str(row)]["Course"])
        
    print(recweek_dict[str(row)]["Course"])

    

BS Legal Management
BS Management
BS Communications Technology
BS IT Entrepreneurship
BS Communications Technology
BS Management
BS Legal Management
BS Management Engineering
BS Management Engineering
BS Communications Technology
BS Legal Management
BS Management Engineering
BS Communications Technology
BS Management
BS IT Entrepreneurship
BS Legal Management
BS IT Entrepreneurship
BS Management
BS IT Entrepreneurship
BS Management Engineering
BS Legal Management
BS IT Entrepreneurship
BS Management
BS Management Engineering
BS Management Engineering
BS Management Engineering
BS Communications Technology
BS Legal Management
BS Legal Management
BS Communications Technology
BS IT Entrepreneurship
BS Management
BS IT Entrepreneurship
BS Legal Management
BS IT Entrepreneurship
BS Management
BS Management
BS Legal Management
BS Management Engineering
BS Legal Management
BS Management
BS Legal Management
BS Management
BS IT Entrepreneurship
BS Management Engineering
BS Management Engineering

BS Communications Technology
BS Legal Management
BS IT Entrepreneurship
BS Communications Technology
BS Management
BS Management Engineering
BS Communications Technology
BS Communications Technology
BS Communications Technology
BS Communications Technology
BS Management Engineering
BS Management
BS Management
BS IT Entrepreneurship
BS Management
BS IT Entrepreneurship
BS Communications Technology
BS Communications Technology
BS Communications Technology
BS IT Entrepreneurship
BS IT Entrepreneurship
BS Legal Management
BS IT Entrepreneurship
BS IT Entrepreneurship
BS Legal Management
BS Management
BS Management
BS Communications Technology
BS Legal Management
BS Management
BS Communications Technology
BS Legal Management
BS IT Entrepreneurship
BS Communications Technology
BS IT Entrepreneurship
BS IT Entrepreneurship
BS Legal Management
BS Communications Technology
BS Management Engineering
BS Communications Technology
BS Management
BS Legal Management
BS IT Entrepreneurship
BS Legal Ma

In [7]:
# Cleaning the Phone Numbers

import re

for row in range (1, len(recweek_dict)+1,1):
    # print(recweek_dict[str(row)]["Phone Number"])
    actual_number = str(((recweek_dict[str(row)])["Phone Number"])).replace(" ", "")
    if len(actual_number) == 10: 
           actual_number = re.sub(r'([0-9]{10})',r'+63\1',actual_number) 
    recweek_dict[str(row)]["Phone Number"] = actual_number
    print(recweek_dict[str(row)]["Phone Number"])


+639183026210
+639272126276
+639182537017
+639187144360
+639274621674
+639186295268
+639173697388
+639172406000
+639182265748
+639183421717
+639277081308
+639277475497
+639179594490
+639275434192
+639172119497
+639181855162
+639271468146
+639275164280
+639279586909
+639175149105
+639279847499
+639276993694
+639173172478
+639171593829
+639188986972
+639177964622
+639271387580
+639171959244
+639176433996
+639177583119
+639186694265
+639177508539
+639272369504
+639174384447
+639177736035
+639184554185
+639278899252
+639278748083
+639278423053
+639279231234
+639183814956
+639186029278
+639185326748
+639176541291
+639182441683
+639178181190
+639174686425
+639186351688
+639178211896
+639279249506
+639188542626
+639184389745
+639178938235
+639182165250
+639172571057
+639279495098
+639277666306
+639188518794
+639179012860
+639186815912
+639183401320
+639275511326
+639189754142
+639174661269
+639276271302
+639175535101
+639174425313
+639277157133
+639174484860
+639172551167
+639184613486
+63927

+639182725703
+639189626954
+639274916684
+639277089046
+639279288064
+639279895381
+639178625481
+639174972907
+639173481333
+639187749374
+639176879040
+639275332835
+639272368391
+639187839530
+639189275880
+639276616951
+639183591983
+639172924179
+639182858534
+639181648297
+639272888492
+639188679643
+639189051512
+639275759786
+639189085902
+639183863965
+639171098620
+639273345450
+639188121222
+639188577371
+639278345521
+639184858289
+639179224155
+639278937520
+639173263403
+639176456607
+639174347543
+639185079807
+639171692137
+639275061238
+639183689133
+639275582114
+639175981160
+639279978860
+639279733481
+639177827480
+639187729716
+639172095036
+639187183375
+639187776708
+639177827638
+639177009826
+639174225632
+639273522048
+639276674402
+639183542271
+639175994743
+639171458973
+639179361237
+639175309570
+639173244108
+639271908210
+639174753924
+639178578466
+639184563730
+639188457547
+639276275354
+639175137001
+639176098149
+639273243799
+639182766108
+63917

In [181]:
# Writing into a CSV File

with open('cleanrecweekdata.csv', 'a', newline='') as csvfile:
    order_writer = csv.writer(csvfile, quoting=csv.QUOTE_MINIMAL)
    order_writer.writerow(["First Name","Last Name","Birth Month", "Birth day", "Birth Year", "Course", "Phone Number"])
    for number in range (1,len(recweek_dict)+1,1):
        order_writer.writerow([recweek_dict[str(number)]["First Name"],recweek_dict[str(number)]["Last Name"],recweek_dict[str(number)]["Month"],recweek_dict[str(number)]["Day"],recweek_dict[str(number)]["Year"],recweek_dict[str(number)]["Course"],recweek_dict[str(number)]["Phone Number"]])
        


# Required Summary Statistics
1. How many people were born in each month?
2. How many people were born on each day?
3. How many people were born in each year?
4. How many people come from each course?
5. How many people have numbers from Globe (0917 or 0927) and Smart (0918)? *Note: it's up to you how to format this output*

Answer these questions in an html file called stats.html. The format of the html is as follows (in template form, so you can use this as format string in your program, though feel free to modify with variable names if you find it easier to do so):


```
<html>
    <body>
         <h2>Summary Statistics</h2>
         <table>
             <tr><th>Stat</th><th>Value</th></tr>
             <tr><td>People born in each month</td><td>{}</td></tr>
             <tr><td>People born on each day</td><td>{}</td></tr>
             <tr><td>People born in each year</td><td>{}</td></tr>
             <tr><td>People coming from each course</td><td>{}</td></tr>
             <tr><td>People having numbers from Globe and Smart</td><td>{}</td></tr>
         </table>
    </body>
</html>
```

In [8]:
# Month Count

month_list = []
for row in range (1, len(recweek_dict)+1,1):
    month_list.append(recweek_dict[str(row)]["Month"])
    
month_tally = {}

month_set = set(month_list)
for month in month_set:
    month_tally[month] = 0
    
for month in month_list:
    month_tally[month] += 1
    
# print(month_tally)

# Arranging the months and corresponding values

# used a number map so as to arrange 1-12 with the given word equivalent, and the month equivalent.
months = dict(one = "January", two = "February", three = "March", four = "April", five = "May", six = "June", seven = "July", eight = "August", nine = "September", ten = "October", eleven = "November", twelve = "December")
num_map = { "one" : 1, "two" : 2, "three": 3, "four": 4, "five": 5, "six": 6, "seven": 7, "eight": 8, "nine": 9,"ten": 10, "eleven": 11, "twelve": 12}

final_month_tally = {}
sorted(months, key = num_map.__getitem__)
for num in sorted(months, key = num_map.__getitem__):
    final_month_tally[months[num]] = month_tally[months[num]]
print(final_month_tally)

{'January': 270, 'February': 232, 'March': 246, 'April': 267, 'May': 273, 'June': 259, 'July': 249, 'August': 241, 'September': 232, 'October': 178, 'November': 277, 'December': 276}


In [9]:
# Day Count

day_list = []
for row in range (1, len(recweek_dict)+1,1):
    day_list.append(recweek_dict[str(row)]["Day"])
    
day_tally = {}

day_set = set(day_list)
for day in day_set:
    day_tally[day] = 0
    
for day in day_list:
    day_tally[day] += 1

# print(day_tally)

# Arranging the days and corresponding values

# Simply used a loop knowing that the values are successive from 1 - 28, with these values as keys. 
final_day_tally = {}
for day in range (1, len(day_tally)+1,1):
    final_day_tally[str(day)] = day_tally[str(day)]

print(final_day_tally)

{'1': 112, '2': 106, '3': 90, '4': 110, '5': 100, '6': 99, '7': 88, '8': 111, '9': 98, '10': 111, '11': 105, '12': 98, '13': 119, '14': 102, '15': 121, '16': 115, '17': 116, '18': 107, '19': 113, '20': 91, '21': 123, '22': 103, '23': 116, '24': 108, '25': 116, '26': 116, '27': 102, '28': 104}


In [11]:
# Year Count

year_list = []
for row in range (1, len(recweek_dict)+1,1):
    year_list.append(recweek_dict[str(row)]["Year"])
    
year_tally = {}

year_set = set(year_list)
for year in year_set:
    year_tally[year] = 0
    
for year in year_list:
    year_tally[year] += 1

#print(year_tally)

# Arranging the years and corresponding values

# used a number map so as to arrange 1-5 with the given word equivalent, and the year equivalent.
years = dict(one= "1997", two= "1998", three= "1999", four= "2000", five= "2001") 
num_map = { "one" : 1, "two" : 2, "three" : 3, "four" : 4, "five" : 5}

final_year_tally = {}
sorted(years, key = num_map.__getitem__)
for num in sorted(years, key = num_map.__getitem__):
    final_year_tally[years[num]] = year_tally[years[num]]
print(final_year_tally)


{'1997': 756, '1998': 746, '1999': 756, '2000': 380, '2001': 362}


In [10]:
# Course Count

course_list = []
for row in range(1, len(recweek_dict)+1, 1):
    course_list.append(recweek_dict[str(row)]["Course"])
    
course_tally = {}

course_set = set(course_list)
for course in course_set:
    course_tally[course] = 0
    
for course in course_list:
    course_tally[course] += 1
    
print(course_tally)

{'BS IT Entrepreneurship': 590, 'BS Management Engineering': 607, 'BS Management': 629, 'BS Legal Management': 585, 'BS Communications Technology': 589}


In [13]:
# Phone Number Count

globe_list = []
smart_list = []

for row in range(1,len(recweek_dict)+1, 1):
    if bool(re.search(r'\+63918[0-9]{7}',recweek_dict[str(row)]["Phone Number"])) == True:
        smart_list.append(recweek_dict[str(row)]["Phone Number"])
    else:
        globe_list.append(recweek_dict[str(row)]["Phone Number"])

provider_tally = {"Smart":len(smart_list),"Globe":len(globe_list)}

print(provider_tally)

{'Smart': 955, 'Globe': 2045}


In [17]:
stats_html = """
<html>
    <body>
         <h1>Summary Statistics</h1>
         <div style = "color : #008000 ; border : 5px solid #008000">
             <h3>People born in each month</h3>
                 <table>
                     <tr><th>Month</th><th>Count</th></tr>
                     {month}
                 </table>
         </div>
         <div style = "color : #008080 ; border : 5px solid #008080">
             <h3>People born on each day</h3>
                 <table>
                     <tr><th>Day</th><th>Count</th></tr>
                     {day}
                 </table>
         </div>
         <div style = "color : #00FFFF ; border : 5px solid #00FFFF">
             <h3>People born on each year</h3>
                 <table>
                     <tr><th>Year</th><th>Count</th></tr>
                     {year}
                 </table>
         </div>
         <div style = "color : #0000FF ; border : 5px solid #0000FF">
             <h3>People coming from each course</h3>
                 <table>
                     <tr><th>Course</th><th>Count</th></tr>
                     {course}
                 </table>
         </div>
         <div style = "color : #000080 ; border : 5px solid #000080">
            <h3>People having numbers from Globe and Smart</h3>
                 <table>
                     <tr><th>Provider</th><th>Count</th></tr>
                     {provider}
                 </table>
         </div>
    </body>
</html>
"""

month_tally_table = ""
for month in final_month_tally:
    table_row = ""
    table_row = table_row + "<tr>"
    table_row = table_row + "<td>" + month + "</td>"
    table_row = table_row + "<td>" + str(final_month_tally[month]) + "</td>"
    table_row = table_row + "</tr>"
    month_tally_table = month_tally_table + table_row


day_tally_table = ""
for day in final_day_tally:
    table_row = ""
    table_row = table_row + "<tr>"
    table_row = table_row + "<td>" + day + "</td>"
    table_row = table_row + "<td>" + str(final_day_tally[day]) + "</td>"
    table_row = table_row + "</tr>"
    day_tally_table = day_tally_table + table_row
    
    
year_tally_table = ""
for year in final_year_tally:
    table_row = ""
    table_row = table_row + "<tr>"
    table_row = table_row + "<td>" + year + "</td>"
    table_row = table_row + "<td>" + str(final_year_tally[year]) + "</td>"
    table_row = table_row + "</tr>"
    year_tally_table = year_tally_table + table_row
    
    
course_tally_table = ""
for course in course_tally:
    table_row = ""
    table_row = table_row + "<tr>"
    table_row = table_row + "<td>" + course + "</td>"
    table_row = table_row + "<td>" + str(course_tally[course]) + "</td>"
    table_row = table_row + "</tr>"
    course_tally_table = course_tally_table + table_row


provider_tally_table = ""
for provider in provider_tally:
    table_row = ""
    table_row = table_row + "<tr>"
    table_row = table_row + "<td>" + provider + "</td>"
    table_row = table_row + "<td>" + str(provider_tally[provider]) + "</td>"
    table_row = table_row + "</tr>"
    provider_tally_table = provider_tally_table + table_row

    
with open('stats.html',"w+") as file:
    content_html = stats_html.format(month = month_tally_table, day = day_tally_table, year = year_tally_table, course = course_tally_table, provider = provider_tally_table)
    file.write(content_html)
    